DateTimeCompareValidator.php

```diff
@@ -154,10 +158,10 @@ protected function validateValue($value)
      * @param mixed $compareValue another value being compared
      * @return boolean whether the comparison using the specified operator is true.
      */
-    protected function compareValues($operator, $value, $compareValue)
+    protected function compareValues($operator, $value, $compareValue, $format)
     {
-    	$dateValue = new \DateTime($value);
-    	$dateCompareValue = new \DateTime($compareValue);
+        $dateValue = empty($format) ? \DateTime::createFromFormat($format, $value) : new \DateTime($value);
+        $dateCompareValue = empty($format) ? \DateTime::createFromFormat($format, $compareValue) : new \DateTime($compareValue);
```

Comment:
You've got your conditionals here backwards. If ```$format``` is empty, you want to use ```new \DateTime($value)```. Reverse the order.